yearn/partners/delegated.py

```diff
@@ -0,0 +1,115 @@
+from collections import defaultdict
+from math import ceil
+
+from brownie import chain
+from toolz import last
+from yearn.events import decode_logs, get_logs_asap
+from yearn.networks import Network
+from yearn.utils import contract
+
+YEARN_PARTNER_TRACKER = contract({
+    Network.Mainnet:"0x8ee392a4787397126C163Cb9844d7c447da419D8",
+    Network.Fantom:"0x086865B2983320b36C42E48086DaDc786c9Ac73B",
+}[chain.id])
+
+class AsOfDict(dict):
+    """
+    Allows you to get the closest previous item.
+    
+    >>> AsOfDict({1: 'a', 2: 'b'})[2]
+    'a'
+    """
+
+    def __getitem__(self, key):
+        return super().__getitem__(last(item for item in sorted(self) if item <= key))
+
+def delegated_deposit_totals():
+    delegated_deposits = defaultdict(lambda: defaultdict(lambda: defaultdict(AsOfDict)))
```

Comment:
how is this `delegated_deposits` populated? I don't see a reference within the loop but just a return at the end of this function